docs/developer/architecture/centralized-cli.rst

```diff
@@ -149,3 +180,217 @@ Cluster Service Administration
+
+This package should be installed on the bootstrap node by default.
+
+Rejected design choice
+""""""""""""""""""""""
+
+- ``Bash`` kubectl plugin: Bash is great to do simple actions but not when
+  you need to do interaction with some API like Kubernetes API or Salt API.
+- ``Python`` kubectl plugin: Python allows us to do complicated actions and
+  great interactions with APIs but interactions between ``Go`` and Kubernetes
+  are much better, with a sample CLI available.
+
+``metalk8sctl`` CLI
+^^^^^^^^^^^^^^^^^^^
+
+Language
+""""""""
+
+``Python`` is chosen as the language for ``metalk8sctl`` for the following
+reasons:
+
+- Ability to interact with Salt directly with Python API.
+- Python installation needed anyway by Salt-minion.
+
+Input and Output
+""""""""""""""""
+
+- Command style:
+
+  .. code-block:: bash
+
+    metalk8sctl <resource> <action>
+
+- Interactive:
+
+  Never.
+
+- Output style:
+
+  Human readable output, do not necessarily need for "machine output" like
+  JSON and YAML.
+
+  The output should display useful information from Salt returns when needed,
+  and in case of error, only show relevant error message(s) from Salt.
+
+.. todo::
+
+  - Define logging informations
+
+Salt interaction
+""""""""""""""""
+
+All Salt interaction should be done using
+`Salt Python API <https://docs.saltstack.com/en/latest/ref/clients/index.html>`_
+and not use the ``salt-call``, ``salt``, ``salt-run`` binary at all.
+
+This Python API allows us to interact with Salt-master directly from the host
+machine as Python API directly acts on the Salt sockets and does not need to
+execute a command inside the Salt-master container.
+
+Deployment
+""""""""""
+
+``metalk8sctl`` should be available from the ISO and also as a System
+Package for Operating System supported by MetalK8s.
+
+This package should be installed on the bootstrap node automatically after
+a fresh install.
+
+As this CLI is used to do the first bootstrap deployment we will need another
+script (likely ``bash``) to configure local repositories and install
+``metalk8sctl`` package with all dependencies.
+
+Rejected design choice
+""""""""""""""""""""""
+
+- ``bash`` MetalK8s CLI: Bash is great to do simple actions but not when you
+  need to do interaction with Salt, Salt API, and Kubernetes API.
+- Do not follow kubectl style for the command (``<action> <resource>``), it
+  does not make sense to regroup command per action as actions are really
+  different and this CLI only manages a few resources.
+
+Implementation Details
+----------------------
+
+Two different projects that can be started in parallel.
+
+First have a simple framework to implement a simple command,
+then each command would update the framework if needed.
+
+Check `Requirements`_ for a full list of commands.
+
+Documentation
+-------------
+
+All command should be documented in the Operational Guide with a reference to
+it when it's needed in the Installation Guide.
```

Comment:
all commands should have a local help if we do not have the guide near us.

Reply:
Indeed, I will add it